src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/editor_state_manager.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  StateComparators,
+  initializeStateManager,
+} from '@kbn/presentation-publishing/state_manager';
+import { OptionsListControlState } from '../../../../common/options_list';
+import { DEFAULT_SEARCH_TECHNIQUE } from './constants';
+
+export type EditorState = Pick<
+  OptionsListControlState,
+  'searchTechnique' | 'singleSelect' | 'runPastTimeout'
+>;
+
+export const editorComparators: StateComparators<EditorState> = {
+  runPastTimeout: 'referenceEquality',
+  searchTechnique: 'referenceEquality',
+  singleSelect: 'referenceEquality',
+};
+
+const defaultEditorState = {
+  searchTechnique: DEFAULT_SEARCH_TECHNIQUE,
+  singleSelect: undefined,
+  runPastTimeout: undefined,
+};
+
+export const initializeEditorStateManager = (initialState: EditorState) => {
+  return initializeStateManager<EditorState>(
+    {
+      ...initialState,
+      searchTechnique: initialState.searchTechnique ?? DEFAULT_SEARCH_TECHNIQUE,
```

Comment:
Do we need to specify `searchTechnique` here again? Won't the defaults populate if searchTechnique is `undefined`? Is the problem that it can be `null` or something similar?

Reply:
searchTechnique could be undefined or not provided. `initializeStateManager` does not set a value to the default value. The default values are only used to iterator over all of the keys.

Reply:
Humm, that might be a bug in the state manager. IMO if there are default values provided in the defaults, the subjects should be created with those defaults.

Reply:
`initializeStateManager` iterates over `const allState = { ...defaultState, ...initialState };` - doesn't that fall back to the default value?

Reply:
We just left it as not providing the default value to avoid changing behavior. Created https://github.com/elastic/kibana/issues/221166 to track

Reply:
right, I will remove and close the issue

Reply:
resolved with 61821da5c1ba76099510257018728ad1ee139ce3